tests/integration/peek-all-test.js

```diff
@@ -89,3 +89,25 @@ test("Calling store.peekAll() after creating a record should return correct data
     assert.equal(get(store.peekAll('person'), 'length'), 1, 'should contain one person');
   });
 });
+
+test("Unloading record that is in a peekAll array", function(assert) {
+  let allPeople = store.peekAll('person');
+
+  run(() => {
+    store.createRecord('person', { name: "Tomster" });
+    store.createRecord('person', { name: "Zoey" });
+  });
+
+  assert.equal(get(store.peekAll('person'), 'length'), 2, 'should contain one person');
+
+  run(() => {
+    allPeople.objectAt(0).unloadRecord();
+
+    assert.equal(get(allPeople, 'length'), 2, 'Unload does not complete until the end of the loop');
```

Comment:
i think we want this to happen before the loop ends to reduce # of wtfs

Reply:
I definitely agree that that would be more ideal. I had an extended conversation with @stefanpenner about that on slack. IIRC his near term objective was to get thing backing to how they were at 2.12, then later down the line rethink some of this stuff. That was several months back so it may have changed. But that was the thought process behind this test.